backend/src/main/java/codezap/template/repository/strategy/FullTextSearchSearchStrategy.java

```diff
@@ -39,6 +39,8 @@ public BooleanExpression matchedKeyword(String trimmedKeyword) {
     private String parseKeyword(String trimmedKeyword) {
         String[] parsedKeywords = trimmedKeyword.split(" ");
         return Arrays.stream(parsedKeywords)
+                .map(keyword -> keyword.replaceAll("[^a-zA-Z0-9가-힣]", ""))
```

Comment:
정규표현식만으로는 의미가 잘 나타나지 않아서 나중에 혼동이 올 것 같습니다...!
두 개의 문자열 중 두 번째 문자열은 비교적 쉽게 의미가 파악되지만 정규표현식은 상수로 빼는 게 나을 것 같은데 초롱 생각 궁금합니다!

Reply:
정규표현식은 컴파일 대상이에요. map을 실행할 때마다 정규표현식을 컴파일하므로, 성능이 떨어져요. 

다음과 같이 컴파일한 표현식을 Pattern 객체로 static 필드에 선언해놓고 사용하는 게 더 좋을 것 같아요. 

```java
private static final Pattern INVALID_CHAR_PATTERN = Pattern.compile("[^a-zA-Z0-9가-힣]");
```
```suggestion
                .map(keyword -> INVALID_CHAR_PATTERN.matcher(keyword).replaceAll(""))
```